Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -324,6 +326,47 @@ public static StringCollectionAssertions Should(this IEnumerable<string> @this)
             return new GenericDictionaryAssertions<TCollection, TKey, TValue>(actualValue);
         }
 
+        /// <summary>
+        /// Returns a <see cref="DataSetAssertions"/> object that can be used to assert the
+        /// current <see cref="DataSet"/>.
+        /// </summary>
+        [Pure]
+        public static DataSetAssertions Should(this DataSet actualValue)
+        {
+            return new DataSetAssertions(actualValue);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="DataTableAssertions"/> object that can be used to assert the
+        /// current <see cref="DataTable"/>.
+        /// </summary>
+        [Pure]
+        public static DataTableAssertions Should<T>(this T actualValue)
```

Comment:
It doesn't work out, because typed `DataTable` objects also implement `IEnumerable<T>`. Using a type parameter with a constraint was the only way I knew of to make it unambiguously have higher priority. Is there another way??

Reply:
aha, it's because of `TypedTableBase<T>`.
I haven't worked with DataSet etc. before, so didn't gave it a thought.

This seems to work, but will fragment the API a bit, like the classes in `Events/`.

```cs
public static DataTableAssertions Should(this DataTable actualValue)
{
    return new DataTableAssertions(actualValue);
}

#if !(NETSTANDARD2_0 || NETCOREAPP2_1)
public static DataTableAssertions Should<T>(this TypedTableBase<T> actualValue)
        where T : DataRow
{
    return new DataTableAssertions(actualValue);
}
#endif
```

Reply:
Ahh okay, yep, there's another more specific type that also permits the use of a constraint. Alrighty :-)

Reply:
Hmm, there is a trade-off. This means that anyone using .NET 4.7+ or .NET Core 3.x+ will need to reference System.Data.DataSetExtensions in order to be able to call `.Should()`, even if they're not working with System.Data types.

```
1>------ Build started: Project: Benchmarks, Configuration: Debug Any CPU ------
1>C:\code\fluentassertions\Tests\Benchmarks\CollectionEqual.cs(28,13,28,46): error CS0012: The type 'TypedTableBase<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Data.DataSetExtensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
1>C:\code\fluentassertions\Tests\Benchmarks\CollectionEqual.cs(34,13,34,31): error CS0012: The type 'TypedTableBase<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Data.DataSetExtensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
1>C:\code\fluentassertions\Tests\Benchmarks\CollectionEqual.cs(40,13,40,31): error CS0012: The type 'TypedTableBase<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Data.DataSetExtensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
1>C:\code\fluentassertions\Tests\Benchmarks\CollectionEqual.cs(46,13,46,31): error CS0012: The type 'TypedTableBase<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Data.DataSetExtensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
1>C:\code\fluentassertions\Tests\Benchmarks\BeEquivalentTo.cs(27,87,27,98): error CS0012: The type 'TypedTableBase<>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Data.DataSetExtensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
1>Done building project "Benchmarks.csproj" -- FAILED.
StopOnFirstBuildError: Build cancelled because project "Benchmarks" failed to build.
Build has been canceled.
```

Thoughts?

Reply:
Also, after making that change, `AcceptApiChanges.ps1` doesn't seem to be working any more. It has simply deleted all the `.approved.txt` files. Investigating...

Reply:
Oh, I see, Approval.Tests has to fail first to write the `.received.txt` files :-)

Reply:
I could be mistaken but it looks like `System.Data.DataSetExtensions` is available pre-.NET Core 3.0 in the form of a NuGet package. Is it preferable to exclude the type from the API in order to avoid adding a NuGet dependency on this package? Or would it be simpler just to add a dependency on it and have the API surface the same with all frameworks? The package claims it can be used with .NET Standard 2.0.

Reply:
No, in that case you can add a conditional dependency for .NET Standard 2.0.

Reply:
I'm confused, it wasn't, overall, a Yes or No question :-)

Are you saying that I should make it reference `System.Data.DataSetExtensions` on as many platforms as possible, whether it's a reference into framework DLLs or a NuGet reference? And then hopefully be able to take out the `#if` around that overload of `Should`? It's okay for FluentAssertions to pull in `System.Data.DataSetExtensions` for all consumers?

Reply:
I think you only need to add that reference for the .NET Standard 2.0 target. For the other targets, you don't need that external NuGet package dependency.

Reply:
I have made this change, and also rebased as the PR branch no longer merged cleanly.